Services/Lekarna.Services.Data/Medicines/MedicinesService.cs

```diff
@@ -0,0 +1,196 @@
+﻿namespace Lekarna.Services.Data
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Lekarna.Data.Common.Repositories;
+    using Lekarna.Data.Models;
+    using Lekarna.Services.Mapping;
+    using Lekarna.Services.Models.Medicines;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.EntityFrameworkCore;
+
+    using static Lekarna.Common.GlobalConstants.Offer;
+
+    public class MedicinesService : IMedicinesService
+    {
+        private readonly IDeletableEntityRepository<Medicine> medicinesRepository;
+        private readonly ITargetsService targetsService;
+        private readonly IDiscountsService discountsService;
+
+        public MedicinesService(
+            IDeletableEntityRepository<Medicine> medicinesRepository,
+            ITargetsService targetsService,
+            IDiscountsService discountsService)
+        {
+            this.medicinesRepository = medicinesRepository;
+            this.targetsService = targetsService;
+            this.discountsService = discountsService;
+        }
+
+        public async Task<string> CreateAsync(string name, decimal price, string offerId, string targetId, string discountId)
+        {
+            var medicine = new Medicine
+            {
+                Name = name,
+                Price = price,
+                OfferId = offerId,
+                TargetId = targetId,
+                DiscountId = discountId,
+            };
+
+            await this.medicinesRepository.AddAsync(medicine);
+            await this.medicinesRepository.SaveChangesAsync();
+            return medicine.Id;
+        }
+
+        public IEnumerable<T> GetAll<T>(int? count = null)
+        {
+            IQueryable<Medicine> query = this.medicinesRepository.All().OrderBy(x => x.Name);
+
+            if (count.HasValue)
+            {
+                query = query.Take(count.Value);
+            }
+
+            return query.To<T>().ToList();
+        }
+
+        public async Task<IEnumerable<T>> GetAllMedicines<T>(string id)
+        {
+            var medicines = await this.medicinesRepository.All()
+                .Where(x => x.OfferId == id)
+                .OrderBy(m => m.Name)
+                .To<T>()
+                .ToListAsync();
+
+            //var x = await this.medicinesRepository.AllAsNoTracking()
+            //  .Select(x => new { x.Name, x.Id, Count = this.medicinesRepository.AllAsNoTracking().Count(y => y.TargetId == x.TargetId) })
+            //  .Where(x => x.Count > 1)
+            //  .ToListAsync();
+
+            return medicines;
+        }
+
```

Comment:
SaveAllFromFile method upload offer with 4 columns from csv file in the database.